### ISSUE TITLE
There maybe an error in the calculation of the best validation loss in train.py

### ISSUE THREAD

**Issue by chrisway613** [nnaisense/bayesian-flow-networks#2]
Thanks for this excellent work! I really the code implement of BFN is very beautiful, both the code structure and style.
But when I take a close look at the training part, I found an error(maybe) in [the calculation of the best validation loss](https://github.com/nnaisense/bayesian-flow-networks/blob/834d89674a039abf7872af8c21f3c7424ed7c803/train.py#L161):
```
best_val_loss = validate(
      cfg=cfg,
      model=model,
      ema_model=ema_model,
      val_dataloader=dataloaders["val"],
      step=step,
      run=run,
      pbar=pbar,
      best_val_loss=best_val_loss,
      checkpoint_root_dir=checkpoint_root_dir,
      accelerator=accelerator,
)
```

because `validate()` always return the current validation loss, I think we should change some way as below:
```
best_val_loss = validate(
      cfg=cfg,
      model=model,
      ema_model=ema_model,
      val_dataloader=dataloaders["val"],
      step=step,
      run=run,
      pbar=pbar,
      best_val_loss=best_val_loss,
      checkpoint_root_dir=checkpoint_root_dir,
      accelerator=accelerator,
)
best_val_loss = min(val_loss, best_val_loss)
```

Am I right? waiting for your reply, thx!

**Comment by flukeskywalker** [comment]
Thanks, you're right! I just pushed a fix.

**Comment by chrisway613** [comment]
Welcome! I really like this work. @flukeskywalker
